### PR TITLE
test mrmatrix

### PR DESCRIPTION
**Pull request by mccalluc**
- No hurry!
- I'll continue to add to this branch so that I can understand mrmatrix better, but if this is the wrong direction, better to figure that out sooner than later.
- Let me know if anyone else should be cc'd?
- master is ahead of develop, which I would usually branch from and merge to.
- If something seems odd, I'll put a TODO inline... probably should be resolved before merge. To begin: 
```python
# TODO: "bounds" kwarg is is not actually used... Is something missing?
```

### CODE REVIEW

**Comment by mccalluc** [comment]
That's all the commits I have here, for now. A few questions in the code as TODOs.

**Comment by pkerpedjiev** [comment]
> Let me know if anyone else should be cc'd?
No one that I know of, it's mostly me that's been working on clodius. @nvictus may be a relevant person to keep in the loop because he has a toon of experience with large matrices.
> master is ahead of develop, which I would usually branch from and merge to.
Since it's mostly been me committing to this repo, having a develop branch seemed excessive. We can switch back to using it to stage changes for a release.

File: test/mrmatrix_test.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
+import unittest
+
+import numpy as np
+from numpy.testing import assert_array_equal
+
+from clodius.tiles.mrmatrix import tileset_info, tiles
+
+class AttrDict(dict):
```
Comment (pkerpedjiev) [inline test/mrmatrix_test.py]: What's the point of this class?

Comment (mccalluc) [reply test/mrmatrix_test.py]: To allow arbitrary attributes on a dict:
```python
>>> plain_dict = {}
>>> plain_dict.some_attr = 'foo'
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'dict' object has no attribute 'some_attr'
>>> class AttrDict(dict): pass
... 
>>> attr_dict = AttrDict()
>>> attr_dict.some_attr = 'foo'
>>> attr_dict.some_attr
'foo'
```

The point is to just to be able to mock up something that looks like an hdf5 object: It has `[]` like a dict, but I also want to add `.attrs`.

Comment (mccalluc) [reply test/mrmatrix_test.py]: Maybe calling it `MockHdf5` would make the intention more clear?

Comment (pkerpedjiev) [reply test/mrmatrix_test.py]: It definitely would.

**Pull request merged**